File: dp_labs_and_solutions/dp_labs_and_solutions/dp_java_labs_and_solutions/interpreter/lab/UnaryMinus.java
// UnaryMinus.java

import java.io.PrintWriter;

public class UnaryMinus extends Unary
{
	public UnaryMinus(Expression aOperand1)
	{
		super(aOperand1);
	}

	public void output(PrintWriter aPrintWriter)
	{
		outputPrefix(aPrintWriter, "-");
	}
}

File: dp_labs_and_solutions/original/dp_labs_and_solutions/dp_matlab_labs_and_solutions/interpreter/lab/UnaryMinus.m
% UnaryMinus.m

classdef UnaryMinus < Unary
	methods
		% constructor:
		function theUnaryMinus = UnaryMinus(aOperand1)
			theUnaryMinus@Unary(aOperand1);
		end

		function output(theUnaryMinus)
			theUnaryMinus.outputPrefix('-');
		end
	end
end
